ui/src/App.js

```diff
@@ -1,21 +1,46 @@
 import React, { Component } from 'react';
+import axios from 'axios';
 import logo from './logo.svg';
 import './App.css';
 
 class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <div className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h2>Welcome to React</h2>
-        </div>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
-    );
-  }
+	state = {
+		info: null,
+		error: null
+	}
+
+	componentDidMount() {
+		axios.get('/info')
+			.then((response) => {
+				this.setState({
+					info: response,
+					error: null,
+				});
+			})
+			.catch((error) => {
+				this.setState({
+					error: error.message
+				});
+			});
+	}
+
+	render() {
+		const { info, error } = this.state;
+		return (
+			<div className="App">
```

Comment:
I think this is just a convention for restricting CSS to a specific component (in this case, the `App` component).  The root element uses the same class name as the component name (`App`). Child elements then use lower class names after a `-` such as `App-header`.

I'm not too fussed about what convention we use, this is just what comes out of the box when generating a sample app using `create-react-app`. It seems a fairly sensible way of namespacing classes to a component if required, and that makes writing tests etc. a bit less hassle.

There are fancier ways of applying classes, such as `react-css-modules` where generated class names contain hash values. This makes writing tests harder though (https://github.com/css-modules/css-modules/issues/194#issuecomment-262625515).

I think once we introduce semantic-ui (or similar), it will reduce the amount of `className` attributes we need to add.